# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/mac/Platform.mm
# Commit: 0441d1553b6b
# Full Hash: 0441d1553b6bbf99d711300680eccd335e0b3545
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2020-07-03 03:56:55
# Regressor Bug: 1649575
# File Overlap Count: 1
# Description:
#   Bug 1649575 - Part 3: Introduce MOXTextMarkerDelegate. r=morgan
#   
#   The delegate is associated with accessible documents (either local or remote).
#   
#   I made a separate protocol for all the text marker stuff as it really is a seperate API.
# ==============================================================================

diff -r 65f8e705d730 -r 0441d1553b6b accessible/mac/Platform.mm
--- a/accessible/mac/Platform.mm	Thu Jul 02 21:24:39 2020 +0000
+++ b/accessible/mac/Platform.mm	Thu Jul 02 21:33:07 2020 +0000
@@ -6,6 +6,8 @@
 
 #import <Cocoa/Cocoa.h>
 
+#import "MOXTextMarkerDelegate.h"
+
 #include "Platform.h"
 #include "ProxyAccessible.h"
 #include "AccessibleOrProxy.h"
@@ -63,6 +65,10 @@
   [wrapper expire];
   [wrapper release];
   aProxy->SetWrapper(0);
+
+  if (aProxy->IsDoc()) {
+    [MOXTextMarkerDelegate destroyForDoc:aProxy];
+  }
 }
 
 void ProxyEvent(ProxyAccessible* aProxy, uint32_t aEventType) {